package.json

```
@@ -270,7 +270,7 @@
 		"build:analyze-bundles": "npm run build -- --webpack-bundle-analyzer",
 		"build:package-types": "node ./bin/packages/validate-typescript-version.js && ( tsc --build || ( echo 'tsc failed. Try cleaning up first: `npm run clean:package-types`'; exit 1 ) ) && node ./bin/packages/check-build-type-declaration-files.js",
 		"prebuild:packages": "npm run clean:packages && lerna run build",
-		"build:packages": "npm run build:package-types && node ./bin/packages/build.js",
+		"build:packages": "npm run --silent build:package-types && node ./bin/packages/build.js",
```

Does it also suppress the error if it happens? Could we run something like `--loglevel error`?

---

@kevin940726: As far as I can tell, it only suppresses NPM errors. For instance, `npm run --silent foobar` yields

```
npm error Missing script: "foobar"
npm error
npm error To see a list of scripts, run:
npm error   npm run

npm error A complete log of this run can be found in: /Users/…/.npm/_logs/….log
```

It's not great but it's good enough for the problem. The thing is that I find it hard to justify spending any more time on this, especially since testing requires checking out a commit predating the type changes, clearing all type data, rebuilding everything, then jumping back to trunk and testing the script — and doing this for the different scenarios. The _proper_ way to do it would be the way I initially drafted it: with a standalone script that lives in `bin/` and orchestrates everything. I can write it by heart and I'm 95% this will work:

```sh
# bin/build-package-types
#########################

#!/bin/sh

# Exit if any command fails.
set -e

# Change to the expected directory.
cd "$(dirname "$0")"
cd ..

echo "Validating TypeScript version..."
node ./bin/packages/validate-typescript-version.js

echo "Building (`tsc --build`)..."
if ! tsc --build; then
    echo 'tsc failed. Try cleaning up first: `npm run clean:package-types`'
    exit 1
fi

echo "Checking build type declaration files..."
node ./bin/packages/check-build-type-declaration-files.js
```

---

> with a standalone script that lives in `bin/` and orchestrates everything.

Sorry I think I missed the convo. Why did we move away from this approach?